### PR TITLE
Download config

### LINKED ISSUE #29: Download config
Add a subcommand for downloading a JSON dump of the autograder configuration.  It would be really useful for committing to version control so that we can check if the AG has drifted.

### PR DESCRIPTION
**Pull request by japplefield**
Closes #29

TODO:

- [x] Tests
- [x] Determine appropriate filename

### CODE REVIEW

File: agiocli/utils.py [hunk 1]
```diff
@@ -439,6 +439,11 @@ def get_group_list(project, client):
     return groups
 
 
+def get_config(project, client):
```
Comment (awdeorio) [inline agiocli/utils.py:442]: Might not need a helper function for this because it's only used once.

**Comment by awdeorio** [comment]
I did some more thinking about the semantics of `agio config` and how it compares to the semantics of the `submissions` and `projects` subcommands.

The `submissions` subcommand shows a JSON blob with submission metadata.  The `submissions --download` flag downloads a copy of a submission's uploaded files.

The `projects` subcommand shows a JSON blob with limited project configuration metadata.

The current `config`  subcommand shows a JSON blob with extensive project configuration metadata.  In particular, what the autograder API calls "ag_test_suites".  It doesn't include the info that the `projects` subcommand produces.

I think the `agio config --download` is confusing because it's very different from `agio submissions --download`.  If we were going to implement a `--download` flag it should probably be to download the autograder support files AKA "instructor_files_needed" e.g. `autograder-999a3f1b.tar.gz`.

## Option 1: New subcommand (current)
Continue to use a new subcommand.  Now that I think about `agio config` is a bit confusing because it's not obvious that we're getting a *project* config.  I mean, it could the config for a course or a single test suite.  Options could include `agio project-config`, `agio project-test-suite-config`, or something else.

## Option 1: projects subcommand flag
We could implement the autograder testcase config dump as `agio projects --verbose`, `agio projects --config`, `agio projects --test-suite-config`, or something along those lines.

WDYT?

**Comment by japplefield** [comment]
I like `agio projects --config`. Would you then add a `--download` and a `--filename` flag?

**Comment by awdeorio** [comment]
> I like `agio projects --config`

OK, let's do it!

> Would you then add a `--download` and a `--filename` flag?

Option 1: do nothing and let the user do `agio projects --config > myconfig.json`

Option 2: `agio projects --output myconfig.json`

Option 3: A global `--output` option much like `--debug` that sends any final output to a file.

Now that I think about it some more, I kinda like Option 3.  We could do the easy route for now (Option 1) and create an issue for Option 3.

**Comment by japplefield** [comment]
Test is kinda underwhelming but given that this route just downloads a long JSON file I think it's appropriate.

**Review by awdeorio — APPROVED** [review]
LGTM after pushing a few nitpicks.   Feel free to merge after taking a look.

**Comment by japplefield** [comment]
This is merged but I think we need to update on Pypi now?

**Comment by awdeorio** [comment]
> This is merged but I think we need to update on Pypi now?

Done!

**Pull request merged**